### PR TITLE
feat: enhance tokens feature control and add error handling

### PR DESCRIPTION
**Pull request by lucas3003**
### Acceptance Criteria

This PR intends to:

1. Add a feature flag to indicate the feature is under maintenance mode.
2. Add a feature flag to turn on/off the navigation bar.
3. Enhance the error handling. New message is presented if status code is different than 200. 

###  Diagram

This diagram assumes that Token feature flag is turned on.
TokensAPI-ErrorHandling drawio

### CODE REVIEW

File: src/components/token/Tokens.js [hunk 1]
```diff
@@ -213,32 +216,40 @@ class Tokens extends React.Component {
...
-                        order={this.state.order}
-                        tableHeaderClicked={this.tableHeaderClicked}
-                        calculatingPage={this.state.calculatingPage}
-                    />
+            <div className="w-100">
+                <div className='col-12'>
+                    <h1>{this.props.title}</h1>
                 </div>
+                {
+                    this.state.maintenanceMode ?
```
Comment (r4mmer) [inline src/components/token/Tokens.js:224]: This ternary inside a ternary is a bit confusing, why not have a separate function to render this part and we can do this conditional rendering with 2 `if`s?

**Review by r4mmer — CHANGES_REQUESTED** [review]

**Review by r4mmer — DISMISSED** [review]

File: src/components/ConditionalNavigation.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const ConditionalNavigation = ({ featureToggle, to, label }) => {
+    return (
+        useFlag(featureToggle) ?
+            <li className="nav-item">
+                <NavLink to={to} exact className="nav-link" activeClassName="active" activeStyle={{ fontWeight: 'bold' }}>{label}</NavLink>
+            </li> : null
+    )
+}
+
+export default ConditionalNavigation;
```
Comment (pedroferreira1) [inline src/components/ConditionalNavigation.js:14]: We should add the component prop types as a documentation.

File: src/components/error/ErrorMessageWithIcon.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        <div className='col-12'>
+            <span>
+                <i className={`fa fa-frown-o`}></i>
+                <strong> {props.message} </strong>
+            </span>
+        </div>
+    )
+}
+
+export default ErrorMessageWithIcon;
```
Comment (pedroferreira1) [inline src/components/error/ErrorMessageWithIcon.js:14]: We should add the component prop types as a documentation.

File: src/components/Navigation.js [hunk 4]
```diff
@@ -68,12 +66,7 @@ class Navigation extends React.Component {
               <li className="nav-item">
                 <NavLink to="/" exact className="nav-link" activeClassName="active" activeStyle={{ fontWeight: 'bold' }}>Transactions</NavLink>
               </li>
-              {
-                shouldRenderCustomTokens ?
-                  <li className="nav-item">
-                    <NavLink to="/tokens" exact className="nav-link" activeClassName="active" activeStyle={{ fontWeight: 'bold' }}>Tokens</NavLink>
-                  </li> : null
-              }
+              <ConditionalNavigation to="/tokens" label="Tokens" featureToggle={`explorer-tokens-${REACT_APP_NETWORK}.rollout`} />
```
Comment (pedroferreira1) [inline src/components/Navigation.js:69]: Should we have in the constants file 

`export UNLEASH_BASE_FEATURE_FLAG = explorer-tokens-${REACT_APP_NETWORK}`

Then we can use this everywhere in the code with `UNLEASH_BASE_FEATURE_FLAG.rollout` or `UNLEASH_BASE_FEATURE_FLAG.maintenance`? What do you think?

File: src/components/token/Tokens.js [hunk 5]
```diff
@@ -51,6 +48,8 @@ class Tokens extends React.Component {
          * loading: Initial loading, when user clicks on the Tokens navigation item
          * isSearchLoading: Indicates if search results are being retrieved from explorer-service
          * calculatingPage: Indicates if next page is being retrieved from explorer-service
+         * error: Indicates if and unexpected error happened when calling the explorer-service
```
Comment (pedroferreira1) [inline src/components/token/Tokens.js:51]: Typo: and -> an

File: src/components/token/Tokens.js [hunk 6]
```diff
@@ -51,6 +48,8 @@ class Tokens extends React.Component {
...
+         * maintenanceMode: Indicates if explorer-service or its downstream services are expecting
```
Comment (pedroferreira1) [inline src/components/token/Tokens.js:52]: are expecting ... ?

File: src/components/token/Tokens.js [hunk 7]
```diff
@@ -63,21 +62,25 @@ class Tokens extends React.Component {
             pageSearchAfter: [],
             loading: true,
             isSearchLoading: false,
-            calculatingPage: false
+            calculatingPage: false,
+            error: false,
+            maintenanceMode: this.props.maintenanceMode
```
Comment (pedroferreira1) [inline src/components/token/Tokens.js:67]: We should add this to proptypes as well

**Review by pedroferreira1 — CHANGES_REQUESTED** [review]

File: src/components/ConditionalNavigation.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+                <NavLink to={to} exact className="nav-link" activeClassName="active" activeStyle={{ fontWeight: 'bold' }}>{label}</NavLink>
+            </li> : null
+    )
+}
+
+/**
+ * label: Navigation link that will render if feature toggle is enabled
+ */
+ConditionalNavigation.propTypes = {
+    label: PropTypes.string.isRequired
```
Comment (pedroferreira1) [inline src/components/ConditionalNavigation.js:19]: You have `to` and `featureToggle` here as well, right?

**Review by pedroferreira1 — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

**Review by r4mmer — APPROVED** [review]

**Pull request merged**